ops/model.py

```diff
@@ -1785,6 +1786,19 @@ def __iter__(self):
     def __getitem__(self, key: Union['Unit', 'Application']) -> 'RelationDataContent':
         return self._data[key]
 
+    def __setitem__(self, key: Union['Unit', 'Application'], value: Dict[str, str]) -> None:
```

Comment:
I think this would be better as a `Mapping[str, str]` than `Dict`. For one thing, that makes the `__getitem__` vs. `__setitem__` types a bit more similar, so I think you would be less likely to expect to get a `dict` from `__getitem__`, and more likely to expect a `Mapping`, which is what you get (although it's more specific and also mutable).